src/main/scala/zio/zmx/diagnostics/ZMXServer.scala

```diff
@@ -160,15 +156,15 @@ private[zmx] object ZMXServer {
       } yield ()
     }
 
-    for {
+    val channel = for {
       addr     <- addressIo
       selector <- Selector.make
       channel  <- server(addr, selector)
       _        <- putStrLn("ZIO-ZMX Diagnostics server started...")
       _        <- serverLoop(selector, channel).forever.forkDaemon
```

Comment:
Will this fiber be terminated when the channel is closed?

Reply:
I've added termination logic (that closes also selector, why is it not Managed as well?), as well as some tests that verify we can successfuly reopen on same port (thus resource dealocation was complete)